lib/ecto/query/inspect.ex

```diff
@@ -239,6 +246,7 @@ defimpl Inspect, for: Ecto.Query do
   end
 
   defp from_sources(%Ecto.SubQuery{query: query}), do: from_sources(query.from)
+  defp from_sources(%Ecto.Query.FromExpr{source: source}), do: from_sources(source)
```

Comment:
I don't like that we are calling `from_sources` recursively. This means if that if we accidentally put a `FromExpr` inside another FromExpr or if we forget to wrap a `from` inside a `FromExpr` at some point, it will go unnoticed.